e2e-tests/tests/test_jolteon_two_chain_commit.py

```diff
@@ -0,0 +1,325 @@
+from time import sleep
+from pytest import mark
+from src.blockchain_api import BlockchainApi, Wallet
+from config.api_config import ApiConfig
+import logging as logger
+import json
+from time import time
+
+
+class TestJolteonTwoChainCommit:
+    """Test cases for Jolteon's 2-chain commit rule (core protocol feature)"""
+    
+    @mark.test_key('JOLTEON-2CHAIN-001')
+    def test_two_chain_commit_rule_verification(self, api: BlockchainApi, config: ApiConfig):
+        """Test Jolteon's 2-chain commit rule
+        
+        This test verifies that blocks are committed when two adjacent certified blocks
+        with consecutive round numbers exist. This is Jolteon's core differentiation
+        from 3-chain HotStuff.
+        
+        Based on Tachyeon test case: Lock and Commit Rules - 2-Chain Commit Rule
+        """
+        logger.info("Starting Jolteon 2-chain commit rule test")
+        
+        # Monitor consensus state over time to observe commit patterns
+        wait_time = 90  # Monitor for 1.5 minutes
+        check_interval = 10  # Check every 10 seconds
```

Comment:
Same. I'm not sure what the getReplicaState returns. But if it has an anchor to the parent block we should utilize that.